doc/design_AsyncNDRegisterAcessor_and_NumericAddressedBackend.dox

```diff
@@ -0,0 +1,158 @@
+namespace ChimeraTK {
+/**
+\page design_AsyncNDRegisterAccessor_NumericAddressed Design: AsyncNDRegisterAccessor
+
+\section  design_AsyncNDRegisterAccessor The AsyncNDRegisterAccessor and the AsyncAccessorManager
+
+The AsyncNDRegisterAccessor is the NDRegisterAccessor implementation for push-type data. It is used
+if accessors with AccessMode::wait_for_new_data are requested. See \ref transferElement_B_8
+"Technical specification: TransferElement B.8" for the requirements it is based on.
+
+The AsyncNDRegisterAccessor is a generic implementation for all backends and contains the following components
+
+- an NDRegisterAccessor<UserType>::Buffer which serves as a send buffer and contains the data value
+  (2D vector of UserType), a VersionNumber and a DataValidity flag.
+- a lock free queue to transport the data of the send buffer
+- functions to activate/deactivate the accessor, and to push data and exceptions into the queue
+- a (synchronous) accessor which is used for write operations (optional)
+
+One AsyncAccessorManager is used per interrupt that can occur. The AsyncAccessorManager has three main functions:
+
+- It serves as a factory for AsyncNDRegisterAccessors
+- It contains a list of created asynchronous accessors and a subscribe/unsubscribe mechanism
+- It provides functions to act on all asynchronous accessors
+
+\subsection AsyncNDRegisterAccessor_details Design decisions and implementation details
+
+- The data transport queue is of type UserType, so the potentially expensive and slow operations (data readout,
+  de-multiplexing, data conversion) are happening only once, before the data is distributed to multiple subscribers.
+- The additional send buffer is necessary because data has to be send through the queue in this container, and usually
```

Comment:
I do not understand this point. Is the "buffer" here the type again, or an instance?

If type is meant: The type is merely needed to aggregate the information for the queue.

If instance is meant: I do not understand from this.

Reply:
With "send buffer" the instance is meant. You have to prepare an instance of the object that is being pushed into the queue because it is swapped with an instance inside the queue when pushing.
When popping the queue this is not necessary if you have set up a continuation. The function that is called there is getting a reference of the received element, and you can extract the data and put it into buffer_2D etc. For sending this is not possible, so here you need an instance of the same type as it is running in the queue, prepare it and then call push.

How do I phrase that?